### PR TITLE
Add API by uploading OpenAPI Specification

### LINKED ISSUE #3061: Add API using Swagger file: parse json and yaml
# User story
As an API owner, I would like to be able to upload or link a Swagger file so that I can add an API to Catalog without manually entering the same data that can already be found from my Swagger file.

# Goal
The goal of this task is to solve basic first steps for adding an API with reading data from Swagger file
* Get familiar with OpenAPI 2.0 specification and understand which fields in the Swagger document can be mapped to these APInf fields
  * Name
  * URL
  * Description
  * lifecycle status
* Find suitable open source NPM packages for
  * parsing YAML
  * parsing JSON
* Figure out how to validate, whether the file is valid Swagger file, valid json, valid yaml

Note: the actual UI implementation of Add API using Swagger to follow later.

### LINKED ISSUE #3181: [API Profile] While API is removing the related ApiDoc instance isn't
Verified on local side. On nightly and other instances it is hard to catch error because a developer doesn't have access to MongoDB

**Steps**
1. Create an API
2. Upload Swagger specification as file
3. Find the related apiDocs in mongoDB and copy value of `fileId` field in the notebook;
4. Remove the API

**Expected result**
1. API is successful removed
2. The related ApiDocs is successful removed
3. The related DocumentationFiles is successful removed.

**Actual result**
1. API is successful removed
2. The related ApiDocs isn't removed. A developer can find one using query `db.apiDocs.find({ fileId: "fileId" });`
3. The related DocumentationFiles isn't removed. A developer can find one using query `db.DocumentationFiles.file.find({ _id: ObjectId("fileId") });`

### LINKED ISSUE #3182: [API profile] two instances to contain data about API Documentation
Verified on local side. 

**Steps**
1. Create a new API for the purity of the experiment
2. Navigate to Documentation tab and click on the button "Manage"
3. Upload file with Swagger specification
screenshot at nov 30 13-13-43
**Result**
A new instance in ApiDocs collection
4. Add a link to external site
5. Click on the button "Save"
**Result**
A new instance in ApiDocs collection
screenshot at nov 30 13-13-05
 
**Expected result**
1. Only one instance with particular "apiId" value
2. The instance contains information about file and link

**Actual result**
1. Two instances have the same value of field "apiId"
2. Instances have different info

### PR DESCRIPTION
**Pull request by marla-singer**
Closes #3061 

Closes #3125 (Saransh's PR)
Closes #3181 (The issue has steps for reproduce)
Closes #3182 (The issue has steps for reproduce)

## Changes
@saransh-dev faced with a problem about how to upload a file into DocumentationFiles collection. In the middle of reviewing his code, I understood that the application has not a good method to uploading a file and working with ApiDocs and DocumenationFiles collections.

As a result, I remade logic about uploading a file and inserting into ApiDocs collection data about the uploaded file. 

## Reviewer checklist
Reviewed by: @saransh-dev @matleppa 

This list is to be completed by the pull request reviewer:
- [ ] Code works as described/expected
- [ ] Code seems to be error free
    - [ ] no browser console errors visible
    - [ ] no server console errors visible
    - [ ] passes CI build
- [ ] Code is written in a way that promotes maintainability
    - [ ] easy to understand
    - [ ] well organized
    - [ ] follows project coding standards and conventions
    - [ ] well documented

### CODE REVIEW

File: apinf_packages/apis/client/import_open_api_spec/import.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+            instance.errorCase(message, fileId);
+          }
+        });
+    }
+  });
+
+  instance.removeDocumentationFile = (fileId) => {
+    // Convert to Mongo ObjectID
+    const objectId = new Mongo.Collection.ObjectID(fileId);
+
```
[the remark this answers is not in the document]
Comment (marla-singer) [reply apinf_packages/apis/client/import_open_api_spec/import.js]: Nope.

**Comment by marla-singer** [comment]
@matleppa Fixed the requested changes

File: apinf_packages/apis/client/import_open_api_spec/import.js [hunk 2]
```diff
@@ -128,14 +129,14 @@ Template.importOpenApiSpecification.helpers({
     return false;
   },
   lifecycleStatus () {
-    // Set allowed lyfecucle status values
-    const allowedStatuses = ['design', 'development', 'testing', 'production', 'deprecated'];
+    // Get allowed lyfecycle status values
+    const allowedStatuses = Apis.schema._schema.lifecycleStatus.allowedValues;
```
Comment (matleppa) [inline apinf_packages/apis/client/import_open_api_spec/import.js]: lyfecycle --> lifecycle

**Comment by marla-singer** [comment]
@matleppa Fixed typo

**Comment by matleppa** [comment]
Issue 3181 (remove related apiDocs when API is removed) - OK
Issue 3182 (duplicate apiDocs when documentation is changed) - OK
Issue 2945 (Design user flow for exporting and importing API configuration) - OK

**Pull request merged**